### PR TITLE
exegol: init at 4.3.1

### PR DESCRIPTION
**Pull request by 0b11stan**
## Description of changes

[Exegol](https://github.com/ThePorgs/Exegol/) is a community-driven hacking environment. Something like Kali Linux but based on containers (docker). It's used more and more by the french and international pentesting and infosec community.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/ex/exegol/package.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  pkgs,
+  fetchPypi,
+}:
+pkgs.python311Packages.buildPythonPackage rec {
+  pname = "Exegol";
+  version = "4.2.5";
+  format = "setuptools";
+  doCheck = false;
+
+  propagatedBuildInputs = with pkgs.python311.pkgs; [
```
Comment (FliegendeWurst) [inline pkgs/by-name/ex/exegol/package.nix:12]:
```suggestion
{
  fetchPypi,
  lib,
  python3,
}:
python3.pkgs.buildPythonApplication rec {
  pname = "Exegol";
  version = "4.2.5";
  format = "setuptools";
  doCheck = false;

  propagatedBuildInputs = with python3.pkgs; [
```

File: pkgs/by-name/ex/exegol/package.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      Exegol is a community-driven hacking environment, powerful and yet
+      simple enough to be used by anyone in day to day engagements. Exegol is
+      the best solution to deploy powerful hacking environments securely,
+      easily, professionally. Exegol fits pentesters, CTF players, bug bounty
+      hunters, researchers, beginners and advanced users, defenders, from
+      stylish macOS users and corporate Windows pros to UNIX-like power users.
+    '';
+    homepage = "https://github.com/ThePorgs/Exegol";
+    changelog = "https://github.com/ThePorgs/Exegol/releases/tag/${version}";
+    license = licenses.gpl3;
```
Comment (FliegendeWurst) [inline pkgs/by-name/ex/exegol/package.nix:38]: gpl3 is imprecise, should be either `gpl3Only` or `gpl3Plus`

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]
Thanks for packaging Exegol. The commit message should be "exegol: init at 4.2.5" as described in CONTRIBUTING.md

File: pkgs/by-name/ex/exegol/package.nix (hunk 2 above)
Comment (0b11stan) [reply pkgs/by-name/ex/exegol/package.nix]: Alright thank you ! I could not find the difference between gpl3Only and gpl3Plus in documentation but comparing with other projects I think that gpl3Only is the right one.

**Comment by 0b11stan** [comment]
Thank you very much for your feedbacks, I hope it's better now.

File: pkgs/by-name/ex/exegol/package.nix (hunk 2 above)
Comment (FliegendeWurst) [reply pkgs/by-name/ex/exegol/package.nix]: The difference is whether the project is licensed under the GPLv3 or the GPLv3 + any later version. Can't be guessed by looking at other packages.

Comment (0b11stan) [reply pkgs/by-name/ex/exegol/package.nix]: Alright, I asked Exegol maintainers, we're going for gpl3Only.

**Review by FliegendeWurst — APPROVED** [review]

**Comment by FliegendeWurst** [comment]
Usually the change to maintainers.nix is split into its own commit. Not sure if it's a hard requirement.

File: pkgs/by-name/ex/exegol/package.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
+{
+  fetchPypi,
+  lib,
+  python3,
+}:
+python3.pkgs.buildPythonApplication rec {
+  pname = "Exegol";
```
[the remark this answers is not in the document]
Comment (0b11stan) [reply pkgs/by-name/ex/exegol/package.nix:7]: Alright, thanks

File: pkgs/by-name/ex/exegol/package.nix [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  version = "4.2.5";
+  format = "setuptools";
+  doCheck = false;
```
[the remark this answers is not in the document]
Comment (0b11stan) [reply pkgs/by-name/ex/exegol/package.nix:10]: You're right, thank you

**Review by 0b11stan — COMMENTED** [review]
Follow @fabaff suggestions

**Comment by 0b11stan** [comment]
I updated the version to match the last release of Exegol and squashed everything into one commit.

File: maintainers/maintainer-list.nix [hunk 5]
```diff
@@ -58,6 +58,12 @@
     See `./scripts/check-maintainer-github-handles.sh` for an example on how to work with this data.
 */
 {
+  _0b11stan = {
+    name = "Tristan Auvinet Pinaudeau";
+    email = "tristan@tic.sh";
+    github = "0b11stan";
+    githubId = 27831931;
+  };
```
Comment (Janik-Haag) [inline maintainers/maintainer-list.nix:66]: this should be it's own commit called: `maintainers: add _0b11stan`

File: pkgs/by-name/ex/exegol/package.nix [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+      the best solution to deploy powerful hacking environments securely,
+      easily, professionally. Exegol fits pentesters, CTF players, bug bounty
+      hunters, researchers, beginners and advanced users, defenders, from
+      stylish macOS users and corporate Windows pros to UNIX-like power users.
+    '';
+    homepage = "https://github.com/ThePorgs/Exegol";
+    changelog = "https://github.com/ThePorgs/Exegol/releases/tag/${version}";
+    license = licenses.gpl3Only;
+    mainProgram = "exegol";
+    maintainers = with maintainers; [_0b11stan];
```
Comment (Janik-Haag) [inline pkgs/by-name/ex/exegol/package.nix:42]:
```suggestion
    maintainers = with maintainers; [ _0b11stan ];
```
small formatting nit.

File: maintainers/maintainer-list.nix (hunk 5 above)
Comment (0b11stan) [reply maintainers/maintainer-list.nix:66]: So sorry, I missed the section in CONTRIBUTING and thought it was not mandatory anymore, I fixed it.

**Review by Janik-Haag — APPROVED** [review]
LGTM
